staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go

```diff
@@ -482,40 +521,51 @@ func TestCreateSecretGeneric(t *testing.T) {
 				secret, err = secretOptions.createSecret()
 			}
 
-			if !test.expectErr && err != nil {
-				t.Errorf("test %s, unexpected error: %v", name, err)
+			if test.expectErr == "" && err != nil {
+				t.Errorf("unexpected error: %v", err)
```

Comment:
If you want you can simplify this one slightly with require as well, and have it return instead of attempt to run the DeepEqual if the assertion fails. My full suggestion is below but this is also fine as-is.

```go
if test.expectErr == "" {
  require.NoError(err)
  if !apiequality.Semantic.DeepEqual(secret, test.expected) {
    t.Errorf("\nexpected:\n%#v\ngot:\n%#v", test.expected, secret)
  }
} else {
  require.Error(t, err)
  require.EqualError(t, err, test.expectErr)
}

```

Reply:
Let's simplify it then, will make a change now